treeherder/log_parser/utils.py

```diff
@@ -209,3 +224,91 @@ def extract_log_artifacts(log_url, job_guid, check_errors):
     )
 
     return artifact_list
+
+
+def extract_struct_log_artifacts(log_url, job_guid, check_errors):
+    """ Generate a summary artifact for the structured log. """
+
+    logger.debug("Parsing structured log at url: {0}".format(log_url))
+
+    ab = StructuredLogArtifactBuilder(log_url)
+    ab.parse_log()
+
+    return [(job_guid, ab.name, "json", json.dumps(ab.get_artifact()))]
+
+
+def post_log_artifacts(project,
+                       job_guid,
+                       job_log_url,
+                       retry_task,
+                       extract_artifacts_cb,
+                       check_errors=False):
+    """Post a list of artifacts to a job."""
+
+    try:
+        credentials = OAuthCredentials.get_credentials(project)
+        req = TreeherderRequest(
+            protocol=settings.TREEHERDER_REQUEST_PROTOCOL,
+            host=settings.TREEHERDER_REQUEST_HOST,
+            project=project,
+            oauth_key=credentials.get('consumer_key', None),
+            oauth_secret=credentials.get('consumer_secret', None),
+        )
+        update_endpoint = 'job-log-url/{0}/update_parse_status'.format(
+            job_log_url['id']
```

Comment:
Don't need to go to the effort of changing it now, but I think we shouldn't worry about sticking to 80 chars - 100 or 120 seems fine to me (in this case it would only be 89 chars). The flake8 config has the line length check disabled.